src/components/NumberInput/NumberInput.jsx

```diff
@@ -10,17 +10,17 @@ import {
 
 class NumberInput extends React.Component {
   handleIncrement = () => {
-    const { maxValue, value, step, onChange } = this.props;
+    const { maxValue, value, step, onChange, disabled } = this.props;
     const prevValue = toNumber(value);
     const nextValue = prevValue + step;
-    if (maxValue >= nextValue) onChange(toString(nextValue));
+    if (!disabled && maxValue >= nextValue) onChange(toString(nextValue));
   }
 
   handleDecrement = () => {
-    const { minValue, value, step, onChange } = this.props;
+    const { minValue, value, step, onChange, disabled } = this.props;
     const prevValue = toNumber(value);
     const nextValue = prevValue - step;
-    if (minValue <= nextValue) onChange(toString(nextValue));
+    if (!disabled && minValue <= nextValue) onChange(toString(nextValue));
```

Comment:
let's decouple presentational guard from inc/dec logic; `onIncrementClick = () => !disabled && this.handleIncrement()`

Reply:
these should have just been made as buttons so that we can pass disabled to them and not even worry about click handling; it would also make them accessible (focusable/pressable with keyboard, etc.)